client/flutter/lib/pages/main_pages/app_tab_router.dart

```diff
@@ -40,6 +41,52 @@ class AppTabRouter extends StatelessWidget {
 
   AppTabRouter(this.tabs, this.navItems);
 
+  @override
+  _AppTabRouterState createState() => _AppTabRouterState(tabs, navItems);
+}
+
+class _AppTabRouterState extends State<AppTabRouter> {
+  CupertinoTabController _controller;
+  FirebaseAnalytics _analytics;
+  int lastTabIndex;
+  @override
+  void initState() {
+    _analytics = FirebaseAnalytics();
+    _controller = CupertinoTabController();
+    lastTabIndex = _controller.index;
+
+    _controller.addListener(
```

Comment:
pardon my unfamiliarity with Dart, but does there need to be any cleanup of listeners?

Reply:
I do not believe so